pkg/sources/job_progress_hook.go

```diff
@@ -143,6 +148,9 @@ func (u *UnitHook) Finish(ref JobProgressRef) {
 			metric.EndTime = snap.EndTime
 			metric.Errors = snap.Errors
 		}
+		if u.finishFunc != nil {
+			go u.finishFunc(*metric)
```

Comment:
Does this _need_ to be done in a goroutine, or is it just for performance? And if it's just for performance, what's your sense of the magnitude of the expected benefit?

Reply:
It's for performance, and my hunch says it could be significant since it's doing IO, but it depends on how frequently this gets called (i.e. how many units are getting scanned).

I'll look into gathering some data.

Reply:
> additional implementation complexity

Are you referring to the locks needed for writing the file? I'm not sure I understand. I was thinking goroutines are light enough that they can be used for async operations like this.